entsoe/parsers.py

```diff
@@ -73,8 +75,11 @@ def parse_generation(xml_text):
         if series is None:
             all_series[ts.name] = ts
         else:
-            series = series.append(ts)
-            series.sort_index()
+            if (series.name == ts.name) & series.index.equals(ts.index):
+                series = series + ts
```

Comment:
Apparently there are cases where, when parsing a generation xml, you encounter a block with the exact same index of the previous block? And then you add them together?
Can you provide some comments/context, perhaps an example so we can verify that this is the correct behaviour?

Reply:
This has to do with incorporating consumption (pumped storage, wind off-shore plants...): by adding timeseries with exactly the same index we guarantee that generation and consumption are assimilated.